.github/workflows/ci_benchmark.yml

```diff
@@ -0,0 +1,127 @@
+### Copied from https://github.com/scikit-image/scikit-image/blob/main/.github/workflows/benchmarks.yml
+
+name: Benchmark
+
+on:
+  pull_request:
+    types: [labeled, synchronize]
+
+permissions:
+  contents: read
+
+jobs:
+  ## This code below is to make sure a new commit on a PR with the label
+  ## retriggers the benchmark, otherwise the label needs to be removed and
+  ## applied again to run the benchmark.
```

Comment:
This could both ways, we only explicitly want to run the benchmarks or we want to continuously runs benchmarks for a specific PR (once labelled). Let me know if there are preferences here.

IMO it makes sense to keep running benchmarks on new commits which are tagged as performance critical.